src/SFA.DAS.PR.Jobs/MessageHandlers/EmployerAccounts/RemovedLegalEntityEventHandler.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.EmployerAccounts.Messages.Events;
+using SFA.DAS.PR.Data;
+using SFA.DAS.PR.Data.Entities;
+using SFA.DAS.PR.Jobs.Models;
+
+namespace SFA.DAS.PR.Jobs.MessageHandlers.EmployerAccounts;
+
+public record PermissionAuditDetails(long accountId, long accountLegalEntityId, long Ukprn, IEnumerable<Operation> operations);
+
+public class RemovedLegalEntityEventHandler(IProviderRelationshipsDataContext _providerRelationshipsDataContext, ILogger<RemovedLegalEntityEventHandler> _logger) : IHandleMessages<RemovedLegalEntityEvent>
+{
+    public const string RemovedLegalEntityEventFailureReason = "Account legal entity does not exist";
+
+    public async Task Handle(RemovedLegalEntityEvent message, IMessageHandlerContext context)
+    {
+        AccountLegalEntity? accountLegalEntity = await _providerRelationshipsDataContext.AccountLegalEntities
+            .Include(a => a.AccountProviderLegalEntities)
+                .ThenInclude(b => b.Permissions)
+            .Include(a => a.AccountProviderLegalEntities)
+                .ThenInclude(b => b.AccountProvider)
+        .FirstOrDefaultAsync(a => a.Id == message.AccountLegalEntityId, context.CancellationToken);
+
+        if (accountLegalEntity is null)
+        {
+            _logger.LogWarning("Legal entity with Id:{LegalEntityId} does not exist", message.LegalEntityId);
+
+            await _providerRelationshipsDataContext.JobAudits.AddAsync(new(
+                    nameof(RemovedLegalEntityEventHandler),
+                    new EventHandlerJobInfo<RemovedLegalEntityEvent>(context.MessageId, message, false, RemovedLegalEntityEventFailureReason)
+                ),
+                context.CancellationToken
+            );
+
+            await _providerRelationshipsDataContext.SaveChangesAsync(context.CancellationToken);
+        }
+        else
+        {
+            IEnumerable<Permission> permissions = accountLegalEntity.AccountProviderLegalEntities.SelectMany(a => a.Permissions);
+
+            IEnumerable<PermissionAuditDetails> auditDetails = accountLegalEntity.AccountProviderLegalEntities.Select(a =>
+                new PermissionAuditDetails(
+                    message.AccountId,
+                    message.AccountLegalEntityId,
```

Comment:
Not a big deal, however AccountId and ALEId being static across all the permissions, I would have rather defined audit details as such...
```
public record AccountLegalEntityRemovedAuditInfo(long AccountId, long AccountLegalEntityId, IEnumerable<PermissionAuditInfo> Permissions); 

public record PermissionAuditInfo(int Ukprn, IEnumerable<int> Operations);
```